crates/utils/src/request.rs

```diff
@@ -85,91 +89,73 @@ pub(crate) struct PictrsFile {
 pub(crate) async fn fetch_pictrs(
   client: &Client,
   image_url: &Url,
-) -> Result<PictrsResponse, LemmyError> {
-  is_image_content_type(client, image_url).await?;
-
-  let fetch_url = format!(
-    "{}/image/download?url={}",
-    Settings::get().pictrs_url(),
-    utf8_percent_encode(image_url.as_str(), NON_ALPHANUMERIC) // TODO this might not be needed
-  );
-
-  let response = retry(|| client.get(&fetch_url).send()).await?;
-
-  let response: PictrsResponse = response
-    .json()
-    .await
-    .map_err(|e| RecvError(e.to_string()))?;
-
-  if response.msg == "ok" {
-    Ok(response)
+) -> Result<Option<PictrsResponse>, LemmyError> {
+  if let Some(pictrs_url) = Settings::get().pictrs_url {
+    is_image_content_type(client, image_url).await?;
+
+    let fetch_url = format!(
+      "{}/image/download?url={}",
+      pictrs_url,
+      utf8_percent_encode(image_url.as_str(), NON_ALPHANUMERIC) // TODO this might not be needed
+    );
+
+    let response = retry(|| client.get(&fetch_url).send()).await?;
+
+    let response: PictrsResponse = response
+      .json()
+      .await
+      .map_err(|e| RecvError(e.to_string()))?;
+
+    if response.msg == "ok" {
+      Ok(Some(response))
+    } else {
+      Err(anyhow!("{}", &response.msg).into())
+    }
   } else {
-    Err(anyhow!("{}", &response.msg).into())
+    Ok(None)
   }
 }
 
 pub async fn fetch_iframely_and_pictrs_data(
   client: &Client,
   url: Option<&Url>,
-) -> (Option<String>, Option<String>, Option<String>, Option<Url>) {
+) -> Result<(Option<IframelyResponse>, Option<Url>), LemmyError> {
```

Comment:
This is the main change from your branch: the Iframely response should not have been a default with a bunch of empty strings, but an Option due to either a missing config, or an error'd iframely fetch. 

A lot of this will probably be moot if / when we replace Iframely with simple internal scraping.

Reply:
That seems to be essentially the same as my version, except that you are making the call sites more verbose by creating the option values there. Functionally the only difference seems to be that you ignore errors from fetch_iframely(). I'm not sure why you're doing that, as they could help debug wrong configuration (eg wrong iframely url set, or iframely url set when iframely isnt installed). Maybe just log the error here?

Reply:
There is a difference between returning a `None` if iframely either isn't set up, or has connection errors, and returning a `IframelyResult { url: bunch of empty strings as defaults }`.

We definitely shouldn't be returning default values if they're actually the result of a missing config or failed fetch.

Reply:
I think i'm a bit confused by this function as it merges pictrs and iframely responses in a weird way. It would probably be clearer with a separate function each.